packages/sit-onyx/src/components/OnyxStepper/OnyxStepper.vue

```diff
@@ -0,0 +1,261 @@
+import { injectI18n } from "../../i18n";
+import OnyxFormElement from "../OnyxFormElement/OnyxFormElement.vue";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import OnyxLoadingIndicator from "../OnyxLoadingIndicator/OnyxLoadingIndicator.vue";
+import OnyxSkeleton from "../OnyxSkeleton/OnyxSkeleton.vue";
+import type { OnyxStepperProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxStepperProps>(), {
+  modelValue: 0,
+  step: 1,
+  stripStep: false,
+  placeholder: "0",
+  disabled: false,
+  readonly: false,
+  loading: false,
+  skeleton: false,
+});
+
+const { t } = injectI18n();
+
+const emit = defineEmits<{
+  /** Emitted when the input value changes. */
+  "update:modelValue": [value: number];
+  /**
+   * Emitted when the current value changes but only when the input is blurred.
+   */
+  change: [value: number];
+  /**
+   * Emitted when the input is focussed.
+   */
+  focus: [];
+  /**
+   * Emitted when the input is blurred.
+   */
+  blur: [];
+  /**
+   * Emitted when the validity state of the input changes.
+   */
+  validityChange: [validity: ValidityState];
+}>();
+
+const { vCustomValidity, errorMessages } = useCustomValidity({ props, emit });
+
+const { densityClass } = useDensity(props);
+
+/**
+ * Current value (with getter and setter) that can be used as "v-model" for the native input.
+ */
+const value = computed({
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
+
+const handleChange = (event: Event) => {
+  const inputValue = (event.target as HTMLInputElement).valueAsNumber;
+  emit("update:modelValue", inputValue);
+};
+
+const handleCounterButton = (event: Event, operation: "add" | "substract") => {
+  event.preventDefault();
+  if (props.disabled || props.readonly || props.loading) return;
+
+  if (operation === "add") {
+    const newValue = value.value + props.step;
+
+    if (props.max && newValue > props.max) return;
+    value.value = value.value + props.step;
+  } else {
+    const newValue = value.value - props.step;
+
+    if (props.min && newValue < props.min) return;
+    value.value = value.value - props.step;
+  }
+};
+</script>
+
+<template>
+  <div v-if="props.skeleton" :class="['onyx-stepper-skeleton', densityClass]">
+    <OnyxSkeleton v-if="!props.hideLabel" class="onyx-stepper-skeleton__label" />
+    <OnyxSkeleton class="onyx-stepper-skeleton__input" />
+  </div>
+  <div v-else :class="['onyx-stepper', densityClass]">
+    <OnyxFormElement v-bind="props" :error-messages="errorMessages">
+      <div class="onyx-stepper__wrapper">
+        <span
+          tabindex="0"
+          role="button"
+          class="onyx-stepper__counter"
+          :class="{ 'onyx-stepper__counter--disabled': props.min && props.min === value }"
+          :aria-label="t('substract')"
+          @click="handleCounterButton($event, 'substract')"
+          @keydown.enter="handleCounterButton($event, 'substract')"
+        >
```

Comment:
Initially I used buttons, but then issues with styling appeared. Because both buttons have the same class, when I hover or click on the plus button the styles are applied on the other button too. Do you know why is this happening?

Reply:
I pushed the changes with native buttons you can check them out
@JoCa96

Reply:
@MajaZarkova This is an issue with our `OnyxFormElement`. I created a PR to address this issue: #1576